tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -290,6 +291,29 @@ def test_slurm_protected_mode(
     _test_recover_from_protected_mode(pending_job_id, pcluster_config_reader, bucket_name, cluster, scheduler_commands)
 
 
+@pytest.mark.usefixtures("region", "os", "instance", "scheduler")
+@pytest.mark.test_slurm_protected_mode_on_cluster_create
+def test_slurm_protected_mode_on_cluster_create(
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    test_datadir,
+    s3_bucket_factory,
+    scheduler_commands_factory,
+):
+    """Test that slurm protected mode triggers head node launch failure on cluster creation."""
+    # Create S3 bucket for pre-install scripts
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "preinstall.sh"), "scripts/preinstall.sh")
+
+    cluster_config = pcluster_config_reader(bucket=bucket_name)
+    cluster = clusters_factory(cluster_config, raise_on_error=False, wait=False)
+    remote_command_executor = _wait_until_protected_mode_failure_count_set(cluster)
+    _test_compute_fleet_status(remote_command_executor, expected_status="PROTECTED")
```

Comment:
There's already an existing function `check_status(cluster, compute_fleet_status="PROTECTED")` to check compute fleet status

Reply:
check_status doesn't work in this case because describe-cluster only returns "unknown" for the compute fleet status when the cluster has not been successfully created.